src/main/java/com/mongodb/spark/sql/connector/config/ReadConfig.java

```diff
@@ -153,6 +157,38 @@
 
   private static final boolean AGGREGATION_ALLOW_DISK_USE_DEFAULT = true;
 
+  /**
+   * Publish Full Document only when streaming.
+   *
+   * <p>Note: Only publishes the actual changed document rather than the full change stream
+   * document. Automatically, sets `{@value STREAM_FULL_DOCUMENT_CONFIG}` so updated documents will
+   * also be included. Also filters the change stream events to include only events with a
+   * "fullDocument" field.
+   *
+   * <p>Configuration: {@value}
+   *
+   * <p>Default: {@value STREAM_PUBLISH_FULL_DOCUMENT_ONLY_DEFAULT}.
+   */
+  public static final String STREAM_PUBLISH_FULL_DOCUMENT_ONLY_CONFIG =
+      "stream.publish.full.document.only";
+
+  private static final boolean STREAM_PUBLISH_FULL_DOCUMENT_ONLY_DEFAULT = false;
+
+  /**
+   * Publish Full Document only when streaming.
+   *
+   * <p>Note: Determines what to return for update operations when using a Change Stream. See: <a
+   * href="https://www.mongodb.com/docs/manual/changeStreams/#lookup-full-document-for-update-operations">
+   * Change streams lookup full document for update operations.</a> for further information.
```

Comment:
As there may be new values in the future, I'd rather not pass any value and leave it to the java driver to issue the command.

Reply:
Added note regarding "updateLookup".

Reply:
> As there may be new values in the future, I'd rather not pass any value and leave it to the java driver to issue the command.

Specifying `STREAM_FULL_DOCUMENT_DEFAULT = "DEFAULT"` does not mean that we have to send it. In fact, because this value is documented by the driver as _"the servers default value"_, the [driver does not send it even if it is used explicitly](https://github.com/mongodb/mongo-java-driver/blob/eeb2681ccf7212dca72aa8f2eca3e8576511b058/driver-core/src/main/com/mongodb/internal/operation/ChangeStreamOperation.java#L432-L433). Regardless of whether there will be more values in the future, the default behavior will have to stay as is for backward compatibility.

Am I missing something from the consideration, or do we actually not have a good reason for introducing a new `""` ("unspecified") value in the Spark connector config?

---
> Added note regarding "updateLookup".

Judging from the implementation of `getStreamFullDocument`, specifically from the line `return FullDocument.valueOf(fullDocument)`, I don't think that `"updateLookup"` is a valid value. As far as I can tell, the only valid values are `"DEFAULT"` and `"UPDATE_LOOKUP"`. If we want to allow users to specify `FullDocument.value`, we should use `FullDocument.fromString`

Reply:
Updated - I still think not setting it is better than explicitly setting it to a value - especially as DRIVERS-2231 clarifies that fullDocument should be optional and not defaulted.

Good catch regarding fromString :+1: